services/QuillLMS/engines/evidence/lib/tasks/plagiarized_responses.rake

```diff
@@ -28,4 +28,90 @@ namespace :plagiarized_responses do
       end
     end
   end
+
+  desc "Reads a CSV of historical responses and applies theoretical plagiarism algorithms to them, then outputs the responses and those plagiarism results into a new CSV for comparison"
+
+  # Intended to operate on data generated via the following query:
+  # SELECT comprehension_prompts.text, entry, feedback_text, feedback_type, comprehension_plagiarism_texts.text AS plagiarism_text
+  # FROM feedback_histories
+  # JOIN comprehension_prompts
+  #     ON feedback_histories.prompt_id = comprehension_prompts.id
+  # JOIN comprehension_prompts_rules
+  #     ON feedback_histories.prompt_id = comprehension_prompts_rules.prompt_id
+  # JOIN comprehension_plagiarism_texts
+  #     ON comprehension_prompts_rules.rule_id = comprehension_plagiarism_texts.rule_id
+
+  task :test, [:input_csv_path] => :environment do |t, args|
+
+    # Levenshtein implementation lifted from https://stackoverflow.com/questions/16323571/measure-the-distance-between-two-strings-with-ruby
+    def levenshtein_distance(src, target, max = nil)
```

Comment:
Googling a bit, Ruby 2.3 ships with this gem:
https://github.com/ruby/did_you_mean:

So you can do this:
```ruby
DidYouMean::Levenshtein.distance(string1, string2)
```
Running this 10K times takes 0.35 seconds, so probably better to use this performance tuned version.
<img width="1305" alt="Screen Shot 2021-11-04 at 4 02 09 PM" src="https://user-images.githubusercontent.com/1304933/140411776-b2376874-8300-4277-8247-34f7acd0f722.png">

Reply:
I was going to point out that one of the reasons we rolled our own was to specify an optional `max` value that would allow us to short-circuit the calculation if the distance was going to be too long.  However, I tried running a benchmark with the built-in, and it looks like the optimization to the algorithm more than makes up for the difference, so using the built-in is faster than having the early bail-out.  I'll update the code to reference the built-in.